core/transaction_pool.go

```diff
@@ -45,10 +46,21 @@ var (
 )
 
 const (
-	maxQueued = 64 // max limit of queued txs per address
+	maxQueued  = 64               // max limit of queued txs per address
+	resendTime = 45 * time.Second // every ~3 blocks
 )
 
-type stateFn func() *state.StateDB
+type (
+	stateFn func() *state.StateDB
+	poolTx  struct {
+		*types.Transaction
+
+		owned     bool
+		addedAt   time.Time
+		sentAt    time.Time
+		postTries int // amount of tx posts
+	}
```

Comment:
I'm almost certain the code will contain race conditions: all the fields are accessed directly, and although most are write once, the `sentAt` (and probably `postTries` also eventually) is both read https://github.com/ethereum/go-ethereum/pull/1669/files#diff-d37f7abd23589257d36b8fa71ac686faR145 and written https://github.com/ethereum/go-ethereum/pull/1669/files#diff-d37f7abd23589257d36b8fa71ac686faR307 concurrently. I think it would be advisable to make this struct thread safe.